linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml

```diff
@@ -23,69 +27,279 @@
+    - ["{{ libvirt_resource_name }}"]
+    - "{{ res_count.stdout }}"
+  loop_control:
+    loop_var: definition
+  remote_user: "{{ res_def['remote_user'] | default('root') }}"
+  delegate_to: "{{ uri_hostname }}"
+  when: node_exists['failed'] is defined and uri_hostname != 'localhost' and res_def['cloud_config'] is defined
+
+- name: "Create directories"
+  file:
+    path: "/tmp/vm-{{ definition[0] }}_{{ definition[1] }}"
+    state: "directory"
+  with_nested:
+    - ["{{ libvirt_resource_name }}"]
+    - "{{ res_count.stdout }}"
+  loop_control:
+    loop_var: definition
+  when: node_exists['failed'] is defined and uri_hostname == 'localhost' and res_def['cloud_config'] is defined 
+
+- name: "Prepare cloud-config/user-data-local"
+  template:
+    src: "templates/cloud-config/user-data-local"
+    dest: "/tmp/vm-{{ definition[0] }}_{{ definition[1] }}/user-data"
+  with_nested:
+    - ["{{ libvirt_resource_name }}"]
+    - "{{ res_count.stdout }}"
+  loop_control:
+    loop_var: definition
+  vars:
+    cloudconfig_users: "{{ res_def['cloud_config']['users'] | default([]) }}"
+  when: res_def['cloud_config'] is defined and node_exists['failed'] is defined and uri_hostname == 'localhost'
+
+- name: "Prepare cloud-config/user-data-remote"
+  template:
+    src: "templates/cloud-config/user-data-remote"
+    dest: "/tmp/vm-{{ definition[0] }}_{{ definition[1] }}/user-data"
+  with_nested:
+    - ["{{ libvirt_resource_name }}"]
+    - "{{ res_count.stdout }}"
+  loop_control:
+    loop_var: definition
+  remote_user: "{{ res_def['remote_user'] | default('root') }}"
+  delegate_to: "{{ uri_hostname }}"
+  when: res_def['cloud_config'] is defined and node_exists['failed'] is defined and uri_hostname != 'localhost'
+
+- name: "Prepare cloud-config/meta-data"
+  template:
+    src: "templates/cloud-config/meta-data"
+    dest: "/tmp/vm-{{ definition[0] }}_{{ definition[1] }}/meta-data"
+  with_nested:
+    - ["{{ libvirt_resource_name }}"]
+    - "{{ res_count.stdout }}"
+  loop_control:
+    loop_var: definition
+  when: node_exists['failed'] is defined and uri_hostname == 'localhost' and res_def['cloud_config'] is defined
+
+- name: "Prepare cloud-config/meta-data remote"
+  template:
+    src: "templates/cloud-config/meta-data"
+    dest: "/tmp/vm-{{ definition[0] }}_{{ definition[1] }}/meta-data"
+  with_nested:
+    - ["{{ libvirt_resource_name }}"]
+    - "{{ res_count.stdout }}"
+  loop_control:
+    loop_var: definition
+  remote_user: "{{ res_def['remote_user'] | default('root') }}"
+  delegate_to: "{{ uri_hostname }}"
+  when: node_exists['failed'] is defined and uri_hostname != 'localhost' and res_def['cloud_config'] is defined
+
+- name: "localhost: Generate ci data cd image for cloud-init when cloud config is defined"
+  command: mkisofs -o /tmp/vm-{{ definition[0] }}_{{ definition[1] }}.iso -V cidata -r -J --quiet /tmp/vm-{{ definition[0] }}_{{ definition[1] }}/user-data /tmp/vm-{{ definition[0] }}_{{ definition[1] }}/meta-data
```

Comment:
Does this mean another requirement for linchpin?

Reply:
@herlo : mkisofs is by default available in fedora , ( not sure about centos but it should be ) . That is why i didnt add it as a dependency.